### PR TITLE
Helper to help components spit out debug logs

### PR DESCRIPTION
**Pull request by MadLittleMods**
Helper to help components spit out debug logs

Part of https://github.com/matrix-org/matrix-react-sdk/pull/8502

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: src/models/room.ts [hunk 1]
```diff
@@ -1216,6 +1216,13 @@ export class Room extends TypedEventEmitter<EmittedEvents, RoomEventHandlerMap>
         return Array.from(this.threads.values());
     }
 
+    /**
+     * @experimental
+     */
+    public getThreadsMap(): Map<string, Thread> {
+        return this.threads;
+    }
```
Comment (MadLittleMods) [inline src/models/room.ts:1224]: The alternative to this is using `getThreads()` but this just returns unhelpful `[[], []]` blank arrays of arrays if you haven't opened the threads yet.

It's useful to see which events are threads so we can return the raw map which has a map of `threadID` -> `Thread`

```json
{
    "$BWFwOdh4QmmUpEHNdjAYXEXM5jue3ZcTtLKhc7YIusw": [],
    "$VknYvjxNw2HKifHp_8C5F_WqDMXBTfdDYYT3-AVnNhY": ["$VknYvjxNw2HKifHp_8C5F_WqDMXBTfdDYYT3-AVnNhY","$c-6O4zVYEpEuXU5bWfqFtJXXKXARxJDn64BO_9HnP4A"]
}
```

File: src/timeline-window.ts [hunk 2]
```diff
@@ -41,10 +41,14 @@ const DEFAULT_PAGINATE_LOOP_LIMIT = 5;
 
 interface IOpts {
     windowLimit?: number;
+    // A label used in the logs to namespace and differentiate the timelines.
+    //ex. main timeline from the thread timeline.
+    contextLabel?: string;
```
Comment (MadLittleMods) [inline src/timeline-window.ts:46]: TODO: Remove as this turned out to not be necessary and we can log everything from `TimelinePanel` in https://github.com/matrix-org/matrix-react-sdk/pull/8502

File: src/models/room.ts (hunk 1 above)
Comment (t3chguy) [reply src/models/room.ts:1224]: Your snippet shows an object string->string[]

But the map is threadId -> Thread

The key in the object id Thread::id so you gain 0 additional information from having the map vs the array of threads

File: src/client.ts [hunk 3]
```diff
@@ -8963,6 +8965,11 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
             },
         );
     }
+
+    public dumpDebugLogs() {
+        logger.debug('matrix-js-sdk: dumping extra debug logs to the console');
+        this.emit(ClientEvent.DumpDebugLogs);
+    }
```
Comment (t3chguy) [inline src/client.ts:8972]: using logs is a really bad mechanism given that it relies on consumers monkey-patching `console` which we don't say anywhere in the generated docs to do

File: src/models/room.ts (hunk 1 above)
Comment (MadLittleMods) [reply src/models/room.ts:1224]: Ahh cool, `thread.id` is perfect 👌

File: src/client.ts (hunk 3 above)
Comment (MadLittleMods) [reply src/client.ts:8972]: This doesn't have to live in `matrix-js-sdk`. What event emitter pattern can we use exclusively in `matrix-react-sdk`? This point is tracked by https://github.com/matrix-org/matrix-react-sdk/pull/8502#discussion_r865555576

**Review by turt2live — CHANGES_REQUESTED** [review]
I don't think we need this at this level - will detail on the react-sdk PR.

**Review by turt2live — CHANGES_REQUESTED** [review]
cc @MadLittleMods : I don't think we need this PR anymore?

**Pull request closed without merging**